### PR TITLE
[x86/Linux] add the process of handling when entry point is funclet.

### PR DESCRIPTION
**Pull request by ragmani**
I'm not sure that the changes are correct even in non-Linux cases.

### CODE REVIEW

**Comment by ragmani** [comment]
cc @parjong

**Comment by hqueue** [comment]
@dotnet-bot please test Tizen armel Cross Debug Build

**Comment by janvorli** [comment]
@ragmani what is the reason for this change? Are we hitting some issue on x86 Linux related to this?

**Comment by ragmani** [comment]
@janvorli This fix is related to issue #10139. 

```
Assert failure(PID 748 [0x000002ec], Thread: 748 [0x02ec]): (value & 0x1) == 0
    File: /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/hash.h Line: 596
    Image: /home/jang/DotNet/enable_x86/work/parjong/overlay.debug/corerun


Program received signal SIGABRT, Aborted.
0xf7fdace0 in ?? ()
(gdb) bt
#0  0xf7fdace0 in ?? ()
#1  0xf790fe51 in TerminateProcess (hProcess=0xffffff01, uExitCode=123456789) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/pal/src/thread/process.cpp:1301
#2  0xf6c32722 in TerminateOnAssert () at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/utilcode/debug.cpp:183
#3  0xf6c33b3a in _DbgBreakCheck (szFile=0xf795963c "/home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/hash.h", iLine=596, szExpr=0xf795967c "(value & 0x1) == 0", fConstrained=0)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/utilcode/debug.cpp:436
#4  0xf6c342fa in _DbgBreakCheckNoThrow (szFile=0xf795963c "/home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/hash.h", iLine=596, szExpr=0xf795967c "(value & 0x1) == 0", fConstrained=0)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/utilcode/debug.cpp:548
#5  0xf6c34977 in DbgAssertDialog (szFile=0xf795963c "/home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/hash.h", iLine=596, szExpr=0xf795967c "(value & 0x1) == 0")
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/utilcode/debug.cpp:735
#6  0xf6cde081 in PtrHashMap::LookupValue (this=0xf62c01c8, key=805327905, pv=0x30005421) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/hash.h:596
#7  0xf6e9c085 in ReadyToRunInfo::GetMethodDescForEntryPoint (this=0xf62c00e0, entryPoint=805327905) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/readytoruninfo.cpp:63
#8  0xf6c60651 in ReadyToRunJitManager::JitCodeToMethodInfo (this=0x807f2b0, pRangeSection=0x80567d8, currentPC=805327923, ppMethodDesc=0x0, pCodeInfo=0x0)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/codeman.cpp:6876
#9  0xf6c59c18 in ExecutionManager::IsManagedCodeWorker (currentPC=805327923) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/codeman.cpp:4303
#10 0xf6c599bd in ExecutionManager::IsManagedCode (currentPC=805327923) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/codeman.cpp:4220
#11 0xf71068fb in IsSafeToHandleHardwareException (contextRecord=0x80dab90, exceptionRecord=0x80dae60) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/exceptionhandling.cpp:5133
#12 0xf77f4cc4 in SEHProcessException (exception=0xffffad70) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/pal/src/exception/seh.cpp:268
#13 0xf77f8088 in common_signal_handler (code=11, siginfo=0x8059e0c, sigcontext=0x8059e8c, numParams=2) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/pal/src/exception/signal.cpp:793
#14 0xf77f7c75 in signal_handler_worker (code=11, siginfo=0x8059e0c, context=0x8059e8c, returnPoint=0x80597b0) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/pal/src/exception/signal.cpp:394
#15 0xf79292db in CallSignalHandlerWrapper0 () at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/pal/src/arch/i386/callsignalhandlerwrapper.S:44
#16 0x30005433 in ?? ()
#17 0x300063ea in ?? ()
#18 0x300064fd in ?? ()
#19 0xf71294e9 in CallDescrWorkerInternal () at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/i386/asmhelpers.S:304
#20 0xf6eb087d in CallDescrWorker (pCallDescrData=0xffffcc68) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/callhelpers.cpp:135
#21 0xf6eb0626 in CallDescrWorkerWithHandler (pCallDescrData=0xffffcc68, fCriticalCall=0) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/callhelpers.cpp:78
#22 0xf6eb25e7 in MethodDescCallSite::CallTargetWorker (this=0xffffcde8, pArguments=0xffffcd90, pReturnValue=0xffffcce0, cbReturnValue=8)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/callhelpers.cpp:645
#23 0xf6e372b3 in MethodDescCallSite::Call_RetArgSlot (this=0xffffcde8, pArguments=0xffffcd90) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/callhelpers.h:433
#24 0xf71865c8 in RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::$_1::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const::{lambda(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*)#1}::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const (this=0xffffceb0, pParam=0xffffcf68)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/assembly.cpp:1851
#25 0xf7182c50 in RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::$_1::operator()(RunMain(MethodDesc*, short, int*, REF<PtrArray>*)::Param*) const (this=0xffffcf60, __EXparam=0xffffcf68)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/assembly.cpp:1865
#26 0xf7182a2e in RunMain (pFD=0xf62c0ed4, numSkipArgs=1, piRetVal=0xffffd024, stringArgs=0xffffd3b8) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/assembly.cpp:1865
#27 0xf7182f5d in Assembly::ExecuteMainMethod (this=0x80c1698, stringArgs=0xffffd3b8, waitForOtherThreads=1) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/assembly.cpp:1947
#28 0xf6c6a6b0 in CorHost2::ExecuteAssembly (this=0x80567a0, dwAppDomainId=1, pwzAssemblyPath=0x80a6d00 u"/home/jang/DotNet/enable_x86/work/parjong/unittest/readytorun/mainv1/mainv1.exe", argc=0, 
    argv=0x0, pReturnValue=0xffffd5bc) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/vm/corhost.cpp:501
#29 0xf6bf1ac1 in coreclr_execute_assembly (hostHandle=0x80567a0, domainId=1, argc=0, argv=0x0, 
    managedAssemblyPath=0x80520fc "/home/jang/DotNet/enable_x86/work/parjong/unittest/readytorun/mainv1/mainv1.exe", exitCode=0xffffd5bc)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/dlls/mscoree/unixinterface.cpp:376
#30 0x0804c60d in ExecuteManagedAssembly (currentExeAbsolutePath=0x8052014 "/home/jang/DotNet/enable_x86/work/parjong/overlay.debug/corerun", 
    clrFilesAbsolutePath=0x80520b4 "/home/jang/DotNet/enable_x86/work/parjong/overlay.debug", 
    managedAssemblyAbsolutePath=0x80520fc "/home/jang/DotNet/enable_x86/work/parjong/unittest/readytorun/mainv1/mainv1.exe", managedAssemblyArgc=0, managedAssemblyArgv=0x0)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:420
#31 0x0804b1cf in corerun (argc=2, argv=0xffffd7b4) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:149
#32 0x0804b2da in main (argc=2, argv=0xffffd7b4) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:161AssemblyArgv=0x0)
    at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:420
#31 0x0804b1cf in corerun (argc=2, argv=0xffffd7b4) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:149
#32 0x0804b2da in main (argc=2, argv=0xffffd7b4) at /home/jang/DotNet/enable_x86/work/parjong/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:161
```

File: src/vm/readytoruninfo.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ MethodDesc * ReadyToRunInfo::GetMethodDescForEntryPoint(PCODE entryPoint)
     }
     CONTRACTL_END;
 
-#ifdef _TARGET_AMD64_
+#if defined(_TARGET_AMD64_) || defined(_TARGET_X86_)
```
Comment (janvorli) [inline src/vm/readytoruninfo.cpp]: This should be enabled only for _TARGET_X86_ only for Unix (FEATURE_PAL). Windows x86 doesn't have this problem.

**Pull request merged**